File: Binary_Tree/104_max_tree_depth/104.cpp
#include "../tree.cpp"
#include <algorithm>
#include <vector>


int maxDepth(BinaryTree *root) {

    if (root == nullptr) {
        return 0;
    }
    return std::max(
        maxDepth(root -> left),
        maxDepth(root -> right)
    ) + 1; 
}


int main() {
    std::vector<int> array = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
    BinaryTree* root = build_tree(array);
    print_tree(root);
    std::cout << " " << std::endl;
    int max_depth = maxDepth(root);
    std::cout << " max depth : " << max_depth << std::endl;
}

File: Binary_Tree/104_max_tree_depth/104.py
import sys
sys.path.append(r'/home/mount/Binary_Tree/')
import tree


def maxDepth(root):
    
    def recur(node):
        
        if node == None:
            return 0
        
        return max(
            recur(node.left),
            recur(node.right)
        ) + 1
    
    return recur(root)


root = tree.build_tree([2, 3, 4, 5, 6, 7, 8, 9, 10, 11])
tree.print_tree(root)
depth = maxDepth(root)
print(f'max depth : {depth}')